src/coreclr/jit/lsraarm64.cpp

```diff
@@ -1997,8 +1997,11 @@ int LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree, int* pDstCou
                 {
                     SingleTypeRegSet candidates = lowVectorOperandNum == 2 ? lowVectorCandidates : RBM_NONE;
 
-                    if (intrin.op2->gtType == TYP_MASK)
+                    if (intrin.op2->OperIs(GT_HWINTRINSIC) &&
+                        (intrin.op2->AsHWIntrinsic()->GetHWIntrinsicId() == NI_Sve_ConvertVectorToMask))
                     {
+                        // Have RBM_ALLMASK candidates only if op2 is VectorToMask
+                        assert(intrin.op2->gtType == TYP_MASK);
```

Comment:
What's the reason for this? It's not clear why `TYP_MASK` would be restricted but `VectorToMask` would not...

Reply:
Without this, we end up assigning `RBM_ALLMASK` candidates for operands of `AbsoluateCompare()`, but they should be assigned `RBM_ALLFLOAT` candidates.